### PR TITLE
CAV-651 - Circuit breaker

### PR DESCRIPTION
**Pull request by mar36tin**

### CODE REVIEW

File: it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala [hunk 1]
```diff
@@ -24,6 +25,8 @@ import play.api.libs.json.Json
 import play.api.libs.ws.WSClient
 import play.api.libs.ws.ahc.AhcCurlRequestLogger
 
+//import scala.util.Random
```
Comment (johno1985) [inline it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala:28]: left by mistake?

File: it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala [hunk 2]
```diff
@@ -35,12 +38,13 @@ class VerifyIntegrationSpec extends AnyWordSpec
 
   "POST /" should {
     "return 202 with valid email address" in {
+      val emailPrefix = RandomStringUtils.randomAlphabetic(4)
```
Comment (johno1985) [inline it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala:41]: was this needed?

File: test/uk/gov/hmrc/cipemail/connectors/CircuitBreakerWrapperSpec.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  trait SetUp {
+    protected implicit val hc: HeaderCarrier = HeaderCarrier()
+    val circuitBreakers = new CircuitBreakerWrapper {
+      override def configCB: CircuitBreakerConfig = CircuitBreakerConfig("Cip Verification", 2, 60.toDuration, 60.toDuration, 60.toDuration, 1, 0)
+
+      override def materializer: Materializer = Materializer(TestActorSystem.system)
+    }
+
+    implicit class IntToDuration(timeout: Int) {
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/connectors/CircuitBreakerWrapperSpec.scala:75]: this block is not needed, see corresponding file in cip-email-verification

**Review by johno1985 — COMMENTED** [review]
minor comments

File: it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala (hunk 1 above)
Comment (mar36tin) [reply it/uk/gov/hmrc/cipemail/VerifyIntegrationSpec.scala:28]: yes thanks

File: test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala [hunk 4]
```diff
@@ -109,14 +114,17 @@ class VerifyConnectorSpec extends AnyWordSpec
   trait SetUp {
 
     implicit val hc: HeaderCarrier = HeaderCarrier()
+    val cbConfigData = CircuitBreakerConfig("", 5, 5.toDuration, 30.toDuration, 5.toDuration, 1, 0)
+    implicit class IntToDuration(timeout: Int) {
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala:118]: block not needed

File: test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala [hunk 5]
```diff
@@ -109,14 +114,17 @@ class VerifyConnectorSpec extends AnyWordSpec
...
+      def toDuration = Duration(timeout, java.util.concurrent.TimeUnit.SECONDS)
+    }
+
+    protected val appConfigMock = mock[AppConfig]
+
+    when(appConfigMock.verificationConfig).thenReturn(CipVerificationConfig(
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala:124]: use idiomatics

Comment (johno1985) [reply test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala:124]: unresolved comment

File: test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala (hunk 4 above)
Comment (johno1985) [reply test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala:118]: unresolved comment

File: test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala (hunk 5 above)
Comment (johno1985) [reply test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala:121]: still unresolved

**Review by johno1985 — CHANGES_REQUESTED** [review]
still one unresolved comment

File: test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala (hunk 5 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipemail/connectors/VerifyConnectorSpec.scala]: resolved

**Review by johno1985 — APPROVED** [review]

**Pull request merged**